### PR TITLE
Bump knative/pkg to 1.10

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This comes with a challenge, which is `k8s.io` dependencies, and thus `replace` is coming
back in the picture (sadly). This will have to be fixed once `knative/pkg` k8s.io
dependencies are up-to-date or at least *compatible* with the rest of the library we use
(`go-containerregistry` is the example today).

This is tracked in knative/pkg here : https://github.com/knative/pkg/issues/2759.

This should help unblocking the rest of the dependabot PRs as well.
It also doesn't change the kubernetes min version if I am not mistaken.

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

/kind misc

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [X] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [X] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [X] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
knative/pkg is updated to v1.10
```

### CODE REVIEW

File: hack/update-deps.sh [hunk 1]
```diff
@@ -22,7 +22,7 @@ source $(git rev-parse --show-toplevel)/vendor/github.com/tektoncd/plumbing/scri
 
 cd ${REPO_ROOT_DIR}
 
-KNATIVE_VERSION="release-1.9"
+KNATIVE_VERSION="release-1.10"
```
Comment (lbernick) [inline hack/update-deps.sh:25]: curious how you figured out which version tag corresponded to the tag release-1.10? When I looked, I saw a branch release-1.10 on the github repo, and a few versions on pkg.go.dev, but couldn't figure out which was which.

**Review by lbernick — APPROVED** [review]

File: hack/update-deps.sh (hunk 1 above)
Comment (vdemeester) [reply hack/update-deps.sh:25]: I am not looking at the tag but at the branches (as doesn't use tags at all). `go get …` and `go mod …` work with branches as well as tags 👼🏼, and not looking at pkg.go.dev at all either 🙃

Comment (lbernick) [reply hack/update-deps.sh:25]: good to know, thanks!

**Review by JeromeJu — APPROVED** [review]
Thanks @vdemeester 
/lgtm

**Pull request merged**